src/content/packages-and-plugins/swift-package-manager/for-app-developers.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Swift Package Manager for app developers
+description: How to use Swift Package Manager for native iOS or macOS dependencies
+---
+
+:::warning
+Flutter is migrating to [Swift Package Manager][] to manage iOS and macOS native
+dependencies.
+Flutter's support of Swift Package Manager is under development.
+The implementation might change in the future.
+Swift Package Manager support is only available on the [`main` channel][].
+Flutter continues to support CocoaPods.
+:::
+
+Flutter's Swift Package Manager integration has several benefits:
+
+1. **Provides access to the Swift package ecosystem**.
+   Flutter plugins can use the growing ecosystem of [Swift packages][]. 
+2. **Simplifies Flutter installation**.
+   Xcode includes Swift Package Manager.
+   You don't need to install Ruby and CocoaPods if your project uses
+   Swift Package Manager.
+
+If you find a bug in Flutter's Swift Package Manager support,
+[open an issue][].
+
+[Swift Package Manager]: https://www.swift.org/documentation/package-manager/
+[`main` channel]: /release/upgrade#switching-flutter-channels
+[Swift packages]: https://swiftpackageindex.com/
+[open an issue]: {{site.github}}/flutter/flutter/issues/new?template=2_bug.yml
+
+{% include docs/swift-package-manager/how-to-enable-disable.md %}
+
+## How to add Flutter to an existing app (add-to-app)
```

Comment:
this was a surprising header in this file, and it was surprising to see this first before "how to add SPM to a project".

Consider:

* How to add SPM to a full Flutter app

<existing doc below>

* How to migrate to SPM for a Flutter add-to-app scenario

Not yet supported, follow this bug, etc...


In this way, the headers rhyme and are clear about their scenarios. WDYT?

Reply:
This is excellent feedback. The guide was misleading: hopefully no one needs to do the manual steps. I restructured the page and added a guide dedicated to the auto migration. I also renamed these headers to align better. See: https://github.com/flutter/website/pull/10827#issuecomment-2231753344